Sladim 264

Jacky20000 opened the pull request:
## Preparations
Please check if the PR fulfills these requirements
- [ ] Readme (How to Install, Run, Use and Configure) wurde aktualisiert falls nötig
- [ ] Das Feature ist dokumentiert
- [ ] Alle nicht getter und setter Methoden sind dokumentiert
- [ ] Es wurden Unit tests für neue Features erstellt
- [ ] Alle Tests müssen erfolgreich sein
- [ ] Mindestens ein Reviewer muss den Pull Request akzeptieren

## :ticket: Description
SLADIM-264 compatible with new backend and search issue by id function

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature. refactoring


* **What is the current behavior?** (You can also link to an open issue here)
was not possible to search for issue by id


* **What is the new behavior (if this is a feature change)?**
its now possible to search for issue by id


* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
search for issue by id


* **Other information**:
a little bit of refactoring

## :lock: Closes
Which issue(s) is (are) being closed by the PR?

Jacky20000 committed:
refactoring

shortened some methods, removed commented out code, hard coded component ID for testing SLADIM-264

Jacky20000 committed:
more refactoring

removed unecessary Http Service in constructor, throw error with string instead of just a string, outsourced the constructor in log receiver service into methods, slimmed down the handlog method in each issue creator SLADIM-264

Jacky20000 committed:
updating doc comments

updated doc comments in the issue creators and renamed a method SLADIM-264

Jacky20000 committed:
added search by issueID

added comments, added search by Issue ID which returns an Issue from the Backend on a post request, new endpoint /issue, change issue format, changed error messages for gql requests SLADIM-264

Jacky20000 committed:
minor changes

SLADIM-264

Jacky20000 committed:
fix test

updated logmessageformat, reverted test SLADIM-264

stiesssh commented on src/IssueFormat.ts:
```diff
@@ -7,11 +7,11 @@ export interface IssueFormat {
+  labels?: string[]; //type will be replaced by ID[] where ID is the ID of the components later on
+  assignees?: string[]; //type will be replaced by ID[] where ID is the ID of the components later on
+  locations?: string[]; //type will be replaced by ID[] where ID is the ID of the components later on
+  startDate?: Date; //type will be replaced by Date
+  dueDate?: Date; //type will be replaced by Date
```
ist das kommentar noch korrekt so? schließlich ist der typ nun schon `Date`. es erschließt sich mir nicht, warum man ihn nochmal ersetzen sollte.

stiesssh commented on src/log-receiver/log-receiver.controller.ts:
```diff
@@ -21,6 +21,17 @@ export class LogReceiverController {
     return this.logRcvService.handleLogMessage(logMessage);
   }
+  /**
+   * the searches for the issue corresponding to the given Issue Id
```
who searches?

stiesssh commented on src/log-receiver/log-receiver.service.ts:
```diff
@@ -103,54 +109,58 @@ export class LogReceiverService implements OnModuleInit {
+     * Handles Log messages and saves the Log in the database with the Issue Id
+     * checks if the Detector is registered or whether the LogMessage has a detector Id
+     *
+     * @param log is the log received by the log receiver controller
```
im code heißt der parameter `logMessage`.

Jacky20000 committed:
minor changes

SLADIM-264

Jacky20000 committed:
component ID now in .env file

a valid component ID now has to be specified in .env file and is not hard coded anymore
